src/main/java/org/dataone/indexer/storage/Storage.java

```diff
@@ -4,17 +4,74 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
 
-import org.dataone.hashstore.exceptions.PidRefsFileExistsException;
-
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dataone.configuration.Settings;
+import org.dataone.hashstore.HashStore;
+import org.dataone.hashstore.HashStoreFactory;
+import org.dataone.hashstore.exceptions.HashStoreFactoryException;
 
 /**
- * The Storage represents the interface to access the objects and system metadata
+ * The HashFileStore implementation of the Storage interface 
  */
-public interface Storage {
+public class Storage {
+
+    private static Log log = LogFactory.getLog(Storage.class);
+    private static Storage instance;
+    private static HashStore hashStore;
+    static {
+        try {
+            instance = new Storage();
+        } catch (IOException e) {
+            log.error(
+                "Dataone-indexer cannot initialize the Storage class since " + e.getMessage());
+        }
+    }
+
+    /**
+     * Private constructor
+     * @throws IOException
+     * @throws HashStoreFactoryException
+     */
+    private Storage() throws HashStoreFactoryException, IOException {
+        String className = Settings.getConfiguration().getString("storage.className");
+        String rootPath = Settings.getConfiguration().getString("storage.hashstore.rootDirectory");
+        if (rootPath == null) {
+            throw new HashStoreFactoryException("HashStorage.constructor - The HashStore root path "
+                        + " is null or blank from the property of storage.hashstore.rootDirectory");
+        }
+        String directoryDepth = Settings.getConfiguration()
+                                               .getString("storage.hashstore.directory.depth", "3");
+        String directoryNameWidth = Settings.getConfiguration()
+                                               .getString("storage.hashstore.directory.width", "2");
+        String fileNameAlgorithm = Settings.getConfiguration()
+                                       .getString("storage.hashstore.fileNameAlgorithm", "SHA-256");
+        String defaultNamespace = Settings.getConfiguration()
+                                         .getString("storage.hashstore.defaultNamespace",
+                                        "https://ns.dataone.org/service/types/v2.0#SystemMetadata");
```

Comment:
Would you please also add these values to [helm/config/dataone-indexer.properties](https://github.com/DataONEorg/dataone-indexer/blob/c9bdd92d8f4cdc70da90a358a208a760a3f1d961/helm/config/dataone-indexer.properties#L43-L42)?

You can assign a default value to each of them, and have a yaml location like this example:

```properties
# Storage properties
storage.className={{ default "org.dataone.hashstore.filehashstore.FileHashStore" .Values.idxworker.storage.hashStoreClassName }}
storage.hashstore.rootDirectory={{ default "./target/hashstore" .Values.idxworker.storage.hashStoreRootDir }}
storage.hashstore.defaultNamespace={{ default "https://ns.dataone.org/service/types/v2.0#SystemMetadata" .Values.idxworker.storage.hashStoreDefaultNamespace }}
# The following three properties must NOT be modified after the hash store is initialized
storage.hashstore.fileNameAlgorithm={{ default "SHA-256" .Values.idxworker.storage.hashStoreAlgorithm }}
storage.hashstore.directory.width={{ default 2 .Values.idxworker.storage.hashStoreDirWidth }}
storage.hashstore.directory.depth={{ default 3 .Values.idxworker.storage.hashStoreDirDepth }}
```